### PR TITLE
Feature clipboard backend

### LINKED ISSUE #6: Clipboard Backend
- Set up storage for clipboard entries
- Add callback function to update the storage

### PR DESCRIPTION
**Pull request by WolfgangDaniel**
Closes #6 & #10

- [x] High Test-Coverage (90%), GUI Code might be an exception
- [x] All Tests passing
- [ ] All files autoformatted (to be discussed)
- [x] Codequality: not many long methods
- [x] Codequality: Encapsulation (classes should be loosely coupled, law of demeter,...)
- [x] Timetracking: Issue Timetracking is complete (Time spent)

### CODE REVIEW

File: RichClipboard.package/RCStorage.class/instance/initialize.st [hunk 1]
```diff
@@ -0,0 +1,4 @@
+initialize-release
+initialize
+	super initialize.
+	self storageBackend: LinkedList new.
```
Comment (jffjhnsn) [inline RichClipboard.package/RCStorage.class/instance/initialize.st:4]: Period in final line should be removed.

File: RichClipboard.package/RichClipboard.class/instance/clipboardChanged.from..st [hunk 2]
```diff
@@ -0,0 +1,3 @@
+events
+clipboardChanged: newClipboardText from: updateSource
```
Comment (jffjhnsn) [inline RichClipboard.package/RichClipboard.class/instance/clipboardChanged.from..st:2]: Perhaps `newClipping: newClipboardText from: updateSource` or `addedNewClipping: ...` would make the purpose of the method clearer? `clipboardChanged` seems a bit too vague.

File: RichClipboard.package/RichClipboard.class/instance/initialize.st [hunk 3]
```diff
@@ -0,0 +1,8 @@
+initialize-release
+initialize
+	super initialize.
+	self storage: RCStorage new.
+	Clipboard default
+		when: #contentChanged
+		send: #clipboardChanged:from:
+		to: self.
```
Comment (jffjhnsn) [inline RichClipboard.package/RichClipboard.class/instance/initialize.st:8]: Period in final line should be removed.

File: RichClipboard.package/StorageTest.class/instance/setUp.st [hunk 4]
```diff
@@ -0,0 +1,3 @@
+running
+setUp
+	self storage: RCStorage new.
```
Comment (jffjhnsn) [inline RichClipboard.package/StorageTest.class/instance/setUp.st:3]: Period in final line should be removed.

File: RichClipboard.package/StorageTest.class/instance/testGetRaisesErrorOnInvalidInput.st [hunk 5]
```diff
@@ -0,0 +1,9 @@
+testing
+testGetRaisesErrorOnInvalidInput
+	| storageItems |
+	storageItems := { 'Test1' . 'Test2' . 'Test3' }.
+	storageItems do: [ :item | self storage add: item ].
+	self should: [self storage get: -1] raise: RCInvalidParameterError.
+	self should: [self storage get: '2'] raise: RCInvalidParameterError.
+	self should: [self storage get: 0.5] raise: RCInvalidParameterError.
+	self should: [self storage get: 4] raise: RCOutOfBoundsError.
```
Comment (jffjhnsn) [inline RichClipboard.package/StorageTest.class/instance/testGetRaisesErrorOnInvalidInput.st:9]: Period in final line should be removed.

File: RichClipboard.package/StorageTest.class/instance/testSizeReturnsNumberOfStoredElements.st [hunk 6]
```diff
@@ -0,0 +1,7 @@
+testing
+testSizeReturnsNumberOfStoredElements
+	| storageItems |
+	storageItems := { 'Test1' . 'Test2' . 'Test3' }.
+	self assert: (self storage size = 0).
+	storageItems do: [ :item | self storage add: item ].
+	self assert: (self storage size = storageItems size).
```
Comment (jffjhnsn) [inline RichClipboard.package/StorageTest.class/instance/testSizeReturnsNumberOfStoredElements.st:7]: Period in final line should be removed.

**Review by jffjhnsn — CHANGES_REQUESTED** [review]
All looks good, just a few extras periods snuck in :smile:

File: RichClipboard.package/RCStorage.class/instance/initialize.st (hunk 1 above)
Comment (martingraf4) [reply RichClipboard.package/RCStorage.class/instance/initialize.st:4]: done

File: RichClipboard.package/RichClipboard.class/instance/clipboardChanged.from..st (hunk 2 above)
Comment (martingraf4) [reply RichClipboard.package/RichClipboard.class/instance/clipboardChanged.from..st:2]: Geändert zu `clipboardChanged: newClipboardContent from: updateSource`

File: RichClipboard.package/RichClipboard.class/instance/initialize.st (hunk 3 above)
Comment (martingraf4) [reply RichClipboard.package/RichClipboard.class/instance/initialize.st:8]: schon gefixt

File: RichClipboard.package/StorageTest.class/instance/setUp.st (hunk 4 above)
Comment (martingraf4) [reply RichClipboard.package/StorageTest.class/instance/setUp.st:3]: done

File: RichClipboard.package/StorageTest.class/instance/testGetRaisesErrorOnInvalidInput.st (hunk 5 above)
Comment (martingraf4) [reply RichClipboard.package/StorageTest.class/instance/testGetRaisesErrorOnInvalidInput.st:9]: done

File: RichClipboard.package/StorageTest.class/instance/testSizeReturnsNumberOfStoredElements.st (hunk 6 above)
Comment (martingraf4) [reply RichClipboard.package/StorageTest.class/instance/testSizeReturnsNumberOfStoredElements.st:7]: done

**Pull request closed without merging**